### PR TITLE
feat(cli): Adding hash command

### LINKED ISSUE #390: Add a command that prints out the template hash
## 📚 Context

Post-build, the image hash is stored in the `.sunodo/image/hash` file in binary format.
For most use cases, however, a hexadecimal format is more useful.
This hash can be displayed in this format by running the following command.

```sh
xxd -c 32 -p .sunodo/image/hash
```

This might not be very intuitive for beginner developers, though.

## ✔️ Solution

Add a new command, e.g. `sunodo hash`, which prints the machine hash in hexadecimal format (also, prefixed by `0x`). The `xxd` utility should not be necessary.

## 📈 Subtasks

-   [ ] Add a `sunodo hash` command
-   [ ] Mention the command in the documentation
-   [ ] Test the command against the traditional method
-   [ ] Test the command when no image has been built yet
-   [ ] Test the command when the image was built but the `hash` file is missing

## 🎯 Definition of Done

-   [ ] The command was added
-   [ ] The command was documented
-   [ ] The command was tested

### PR DESCRIPTION
**Pull request by endersonmaia**
related to #399 
fixes #390

### CODE REVIEW

File: apps/cli/src/commands/hash.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    public async run() {
+        // read hash of the cartesi machine snapshot, if one exists
+        const hashPath = path.join(".sunodo", "image", "hash");
+
+        if (fs.existsSync(hashPath)) {
+            return this.getHash(hashPath);
+        }
+
+        this.error("Hash file not found");
```
Comment (tuler) [inline apps/cli/src/commands/hash.ts:35]:
```suggestion
        this.error("Cartesi machine snapshot not found, run 'sunodo build'");
```

File: apps/cli/src/commands/hash.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        "Converts the binary generated by the build command to hexadecimal and prints out the result to console";
+
+    public static enableJsonFlag = true;
+
+    private async getHash(hashPath: string) {
+        const hash = fs.readFileSync(hashPath).toString("hex");
+
+        if (!this.jsonEnabled()) {
+            process.stdout.write(
+                `${chalk.green("?")} Cartesi Machine templateHash ${chalk.cyan(hash)}\n`,
```
Comment (tuler) [inline apps/cli/src/commands/hash.ts:19]:
```suggestion
                `${chalk.green("?")} Cartesi machine templateHash ${chalk.cyan(hash)}\n`,
```

File: README.md [hunk 3]
```diff
@@ -12,6 +12,7 @@ It helps developers to:
 -   `deploy` applications to a cloud provider
 -   `monitor` application already running on a cloud provider
 -   `doctor` verify the minimal requirements for the Sunodo CLI commands
+-   `hash` prints out Cartesi Machine templateHash generated by the build command
```
Comment (endersonmaia) [inline README.md:15]:
```suggestion
-   `hash` prints out Cartesi machine templateHash generated by the build command
```

**Comment by tuler** [comment]
Squash commits

**Comment by endersonmaia** [comment]
> Squash commits

We can now squash and merge right from GitHub web interface, I'll do after approval.

Because it's a single commit.

**Review by tuler — APPROVED** [review]

**Pull request merged**